scripts/get-venue-users-reports.ts

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env node -r esm -r ts-node/register
+import { initFirebaseAdminApp, makeScriptUsage } from "./lib/helpers";
+import admin from "firebase-admin";
+import * as path from "path";
+import * as fs from "fs";
+import { FormatterOptionsArgs, Row, writeToStream } from "@fast-csv/format";
+
+type CsvFileOpts = {
+  headers: string[];
+  path: string;
+};
+
+class CsvFile {
+  static write(
+    stream: NodeJS.WritableStream,
+    rows: Row[],
+    options: FormatterOptionsArgs<Row, Row>
+  ): Promise<void> {
+    return new Promise((res, rej) => {
+      writeToStream(stream, rows, options)
+        .on("error", (err: Error) => rej(err))
+        .on("finish", () => res());
+    });
+  }
+
+  private readonly headers: string[];
+
+  private readonly path: string;
+
+  private readonly writeOpts: FormatterOptionsArgs<Row, Row>;
+
+  constructor(opts: CsvFileOpts) {
+    this.headers = opts.headers;
+    this.path = opts.path;
+    this.writeOpts = { headers: this.headers, quote: true };
+  }
+
+  create(rows: Row[]): Promise<void> {
+    return CsvFile.write(fs.createWriteStream(this.path), rows, {
+      ...this.writeOpts,
+    });
+  }
+}
+
+const usage = makeScriptUsage({
+  description: "Bulk upload events from a spreadsheet ",
+  usageParams: "PROJECT_ID CREDENTIAL_PATH",
+  exampleParams: "co-reality-staging add-event-by-csv.example.csv key.json",
+});
+
+const [projectId, credentialPath] = process.argv.slice(2);
+
+if (!credentialPath) {
+  usage();
+}
+
+initFirebaseAdminApp(projectId, {
+  credentialPath: credentialPath
+    ? path.resolve(__dirname, credentialPath)
+    : undefined,
+});
+
+const runner = async () => {
+  const allUsers: admin.auth.UserRecord[] = [];
+  let nextPageToken: string | undefined;
+  const { users, pageToken } = await admin.auth().listUsers(10);
+
+  allUsers.push(...users);
+  nextPageToken = pageToken;
+
+  while (nextPageToken) {
+    const { users, pageToken } = await admin
+      .auth()
+      .listUsers(1000, nextPageToken);
+    allUsers.push(...users);
+    nextPageToken = pageToken;
+  }
```

Comment:
We shouldn't be reading all users within the auth database like this. There are better patterns in `scripts/get-badges.ts` script for how to handle this and only get the explicit needed users.

Reply:
Why are you listing `10` users at first, and then `1000`?